src/triggers/projectUpdateV2.ts

```diff
@@ -0,0 +1,182 @@
+
+interface ProjectUpdate {
+  id: string;
+  body: string;
+  url: string;
+  createdAt: Date;
+  updatedAt: Date;
+  editedAt: Date | null;
+  health: string;
+  project: {
+    id: string;
+    name: string;
+  };
+  user: {
+    id: string;
+    name: string;
+    email: string;
+  };
+}
+
+interface ProjectUpdatesResponse {
+  data: {
+    projectUpdates: {
+      nodes: ProjectUpdate[];
+    };
+  };
+}
+
+type EventType = "create" | "update";
+
+const subscribeHook = (eventType: EventType) => async (z: ZObject, bundle: Bundle) => {
+  z.console.log("bundle", bundle);
+  const data = {
+    url: bundle.targetUrl,
+    inputData:
+      bundle.inputData && Object.keys(bundle.inputData).length > 0
+        ? pick(bundle.inputData, ["creatorId", "projectId", "teamId"])
+        : undefined,
+  };
+
+  const webhookType = eventType === "create" ? "createProjectUpdate" : "updateProjectUpdate";
+  return z
+    .request({
+      url: `https://client-api.linear.app/connect/zapier/subscribe/${webhookType}`,
+      method: "POST",
+      body: data,
+    })
+    .then((response) => response.data);
+};
+
+const getProjectUpdateList = () => async (z: ZObject, bundle: Bundle) => {
+  const variables: Record<string, string> = {};
+  const variableSchema: Record<string, string> = {};
+  const filters: unknown[] = [];
+  if (bundle.inputData.creatorId) {
+    variableSchema.creatorId = "ID";
+    variables.creatorId = bundle.inputData.creatorId;
+    filters.push({ user: { id: { eq: new VariableType("creatorId") } } });
+  }
+  if (bundle.inputData.teamId) {
+    variableSchema.teamId = "ID";
+    variables.teamId = bundle.inputData.teamId;
+    filters.push({ project: { accessibleTeams: { id: { eq: new VariableType("teamId") } } } });
+  }
+  if (bundle.inputData.projectId) {
+    variableSchema.projectId = "ID";
+    variables.projectId = bundle.inputData.projectId;
+    filters.push({ project: { id: { eq: new VariableType("projectId") } } });
+  }
+  const filter = { and: filters };
+
+  const jsonQuery = {
+    query: {
+      __variables: variableSchema,
+      projectUpdates: {
+        __args: {
+          first: 25,
+          filter,
+        },
+        nodes: {
+          id: true,
+          body: true,
+          url: true,
+          createdAt: true,
+          updatedAt: true,
+          editedAt: true,
+          health: true,
+          project: {
+            id: true,
+            name: true,
+          },
+          user: {
+            id: true,
+            name: true,
+            email: true,
+          },
+        },
+      },
+    },
+  };
+  const query = jsonToGraphQLQuery(jsonQuery);
+
+  const response = await z.request({
+    url: "https://api.linear.app/graphql",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: bundle.authData.api_key,
+    },
+    body: {
+      query,
+      variables,
+    },
+    method: "POST",
+  });
```

Comment:
Want to refactor this and reuse? Feels very copypasta

Reply:
Done